meals.js

```diff
@@ -0,0 +1,46 @@
+let User = {
+  name: 'User McGee',
+  age: 103,
+  calorieTarget: 2000, //target daily calorie intake
+  meals: [
+    { title: 'breakfast',
+      date: '2016-01-20',
+      description: 'eggs',
+      calories: 500},
+    { title: 'lunch',
+      date: '2016-01-20',
+      description: 'sandwich',
+      calories: 400},
+    { title: 'dinner',
+      date: '2016-01-20',
+      description: 'steak',
+      calories: 1000}
+  ],
+  caloriesEatenOn: function(date){
+    let totalCalories = 0;
+    for(i=0; i < this.meals.length; i++){
+      if(date === this.meals[i].date){
+        totalCalories += this.meals[i].calories;
+      }
+    }
+    return totalCalories;
+  },
+  avgDailyCalories: function(){
```

Comment:
I like how you use the pattern @gaand introduced for finding unique strings.

I challenge you to extract the lines of code that only deal with uniqueness into a new method, named `uniqueDays`. Then you can call `uniqueDays` so that line 38 goes away and line 39 becomes `return totalCalories / uniqueDays`.